src/models/__tests__/profile.test.ts

```diff
@@ -184,147 +185,151 @@ describe("Profile codec", () => {
   });
 });
 
-/*
 describe("createProfile", () => {
-  it("should create a new profile", async () => {
-    const clientMock: any = {
-      createDocument: jest.fn((_, newDocument, __, cb) => {
-        cb(undefined, {
-          ...newDocument,
-          _self: "self",
-          _ts: 123
-        });
-      })
-    };
+  const newProfile: NewProfile = {
+    kind: "INewProfile",
+    ...aRawProfile,
+    servicePreferencesSettings: {
+      mode: ServicesPreferencesModeEnum.LEGACY,
+      version: -1
+    }
+  };
 
-    const model = new ProfileModel(clientMock, profilesCollectionUrl);
+  it("should create a new profile", async () => {
+    const containerMock = ({
+      items: {
+        create: jest.fn().mockReturnValue(
+          Promise.resolve({
+            resource: aRetrievedProfile
+          })
+        )
+      }
+    } as unknown) as Container;
 
-    const newProfile: Profile = {
-      fiscalCode: aFiscalCode
-    };
+    const model = new ProfileModel(containerMock);
 
-    const result = await model.create(newProfile, newProfile.fiscalCode);
+    const result = await model.create(newProfile)();
 
-    expect(clientMock.createDocument).toHaveBeenCalledTimes(1);
-    expect(clientMock.createDocument.mock.calls[0][1].kind).toBeUndefined();
-    expect(clientMock.createDocument.mock.calls[0][2]).toHaveProperty(
-      "partitionKey",
-      aFiscalCode
+    expect(containerMock.items.create).toHaveBeenCalledTimes(1);
+    expect(containerMock.items.create).toHaveBeenCalledWith(
+      expect.objectContaining({
+        kind: newProfile.kind,
+        fiscalCode: newProfile.fiscalCode
+      }),
+      expect.anything()
     );
-    expect(isRight(result)).toBeTruthy();
-    if (isRight(result)) {
-      expect(result.value.fiscalCode).toEqual(newProfile.fiscalCode);
-      expect(result.value.id).toEqual(`${aFiscalCode}-${"0".repeat(16)}`);
-      expect(result.value.version).toEqual(0);
-      expect(result.value.isTestProfile).toEqual(false);
+    expect(E.isRight(result)).toBeTruthy();
+    if (E.isRight(result)) {
+      expect(result.right.fiscalCode).toEqual(newProfile.fiscalCode);
```

Comment:
We are checking that values returned from a mocked function that is returning a mock are equal to those into the mock... Is this really useful? If yes, can we avoid rewriting values such as `${aFiscalCode}-${"0".repeat(16)}` and take them directly from the mock?

Reply:
IMHO just checking that functions have been called and the result is Right as we expect is enough.

Reply:
Little note: if `right` is the return of the `create` function with additional processing/decoding, then is perfectly right to check all the value with those in `aRetrievedProfile`.

Reply:
decoding is taking place after the resource is returned. I'll check if the returned resource is strict equal to `aRetrievedProfile`. thanks